commons/imageid_test.go

```diff
@@ -331,53 +326,37 @@ func TestValidateInvalid(t *testing.T) {
 		Tag:  "feature",
 	}
 
-	if iid.Validate() {
-		t.Fatal("expecting failure, bad user")
-	}
+	c.Assert(iid.Validate(), Equals, false)
 }
 
-func TestValidateValid(t *testing.T) {
+func (s *TestCommonsSuite) TestValidateValid(c *C) {
 	iid := &ImageID{
 		Repo: "sierramadre",
 		Tag:  "543c56d1-2510-cd37-c0f4-cab544df985d",
 	}
 
-	if !iid.Validate() {
-		t.Fatal("expecting success: ", iid.String())
-	}
+	c.Assert(iid.Validate(), Equals, true)
 }
 
 type ImageEqualsTest struct {
 	id1, id2 string
 	expected bool
 }
 
-func DoImageEqualsTest(t *testing.T, tests []ImageEqualsTest) {
-	for i, tt := range tests {
+func doImageEqualsTest(c *C, tests []ImageEqualsTest) {
+	for _, tt := range tests {
 		iid1, err := ParseImageID(tt.id1)
-		if err != nil {
-			t.Fatalf("error parsing %s on %d: %s", tt.id1, i, err)
-		}
+		c.Assert(err, IsNil)
 
 		iid2, err := ParseImageID(tt.id2)
-		if err != nil {
-			t.Fatalf("error parsing %s on %d: %s", tt.id2, i, err)
-		}
-
-		expected := "match"
-		if !tt.expected {
-			expected = "mismatch"
```

Comment:
I think the asserts I added at lines 354-355 cover the mismatch cases.  In the previous impl, the stack var `expected` was simply used in the test-failure messages.